fdbserver/workloads/KVStoreTest.actor.cpp

```diff
@@ -98,6 +102,52 @@ class TestHistogram {
 	T sumSQ;
 	uint64_t N;
 };
+struct ClearRangeResultWriter : public IReadRangeResultWriter {
+	// Result writer used for testing range reads with clear ranges
+	int rowLimit, byteLimit;
+	Standalone<RangeResultRef> result;
+	Key clearStart, clearEnd;
+	bool forward = true;
+
+	ClearRangeResultWriter(int rowLimit, int byteLimit) : rowLimit(rowLimit), byteLimit(byteLimit) {
+		if (this->rowLimit < 0) {
+			forward = false;
+			this->rowLimit = -this->rowLimit;
+		}
+	}
+
+	void setClearRange(Key clearStart, Key clearEnd) {
+		this->clearStart = clearStart;
+		this->clearEnd = clearEnd;
+	}
+
+	void clearResults() { result = Standalone<RangeResultRef>(); }
+
+	std::variant<bool, KeyRef> operator()(Optional<KeyValueRef> kv) override {
+		/*
+		    - What this writer does is pass through any values which are not part of a clear range
+		    - If we encounter a key that falls within a clear range then we pass either the begining or the end
+		      of the range (depending on the sign of the limit)
+		*/
+		if (rowLimit <= 0 || byteLimit <= 0)
+			return false;
+		if (!kv.present())
+			return false;
+		if (clearStart.toString() != "" && clearEnd.toString() != "" && kv.get().key >= clearStart &&
+		    kv.get().key < clearEnd) {
+			if (forward)
+				return clearEnd;
+			else
+				return clearStart;
+		}
+		rowLimit--;
+		result.push_back(result.arena(), kv.get());
+		byteLimit -= (sizeof(KeyValueRef) + kv.get().expectedSize());
```

Comment:
Just curious, why is the size of a KeyValueRef included in the byte limit? Is it not included in the expected size?

Reply:
Yeah if you look at the calculation here: https://github.com/apple/foundationdb/blob/master/fdbclient/FDBTypes.h#L395 it never takes into account the size of the keyref (only the key and value)